### PR TITLE
GraphQL Documentation

### LINKED ISSUE #400: Documentation: GraphQL API
# Summary

As a developer I would like to be able to see documentation on how to access and use the `graphql` api.

# Acceptance Criteria

* add in an `Extend` top level section and place it between `Use IRIDA Next` and `Contribute`
* within the `Extend` section add a `GraphQL API` section, be inspired by 'https://docs.gitlab.com/ee/api/graphql/'
* make sure to include note on:
  * how to access  `graphiql`
  * how to authenticate using `personal_access_tokens`
  * include documentation on token scopes

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
## What does this PR do and why?
Adding documentation for graphql

Also fixing formatting errors in docs files touched by the PR

Fixes #400 

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
Docs can be built locally with instructions found in `docs-site/README.md`

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: docs-site/docs/extend/access_tokens.png
Comment (ericenns) [inline docs-site/docs/extend/access_tokens.png:1]: **nitpick (blocking):** Can you remove this screenshot, we are staying away from including screenshots in our documentation so that it doesn't get out of date and instead just use descriptive steps on how to perform the action. See an example here https://phac-nml.github.io/irida-next/docs/user/organization/members/group-and-project-members#add-members-to-a-project

File: docs-site/docs/extend/edit_profile.png
Comment (ericenns) [inline docs-site/docs/extend/edit_profile.png:1]: **nitpick (blocking):** Remove this screenshot as well.

File: docs-site/docs/extend/graphql.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+</td>
+</tr>
+</table>
+
+## Authentication with Personal Access Tokens
+
+To use the GraphQL API with [any of the GraphQL Clients](https://graphql.org/code/#graphql-clients), you will need to generate a Personal Access Token.
+
+### Navigate to the Access Tokens screen
```
Comment (ericenns) [inline docs-site/docs/extend/graphql.md:169]: **nitpick (blocking):** Please remove the screenshots and just use descriptive text instead, like https://phac-nml.github.io/irida-next/docs/user/organization/members/group-and-project-members#add-members-to-a-project

File: docs-site/docs/intro.md
Comment (ericenns) [inline docs-site/docs/intro.md:1]: **nitpick (blocking):** Can you undo this change, I want it to show up as `Intro` on the sidebar which is what this original file did.

File: docs-site/docs/extend/graphql.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+It is important that you do not share this token with anyone else as it is directly tied to your account.
+
+### Using Token for Sessionless Authentication
+
+Tokens generated by IRIDA Next are `Basic Authentication` tokens. They are tied to your email.
+
+This uses [Basic HTTP Authentication](https://datatracker.ietf.org/doc/html/rfc7617), in the form of $USEREMAIL:$ACCESSTOKEN that is Base64 encoded.
+
+e.g. `admin@email.com:yK1euURqVRtQ1D-3uKsW`` becomes `YWRtaW5AZW1haWwuY29tOnlLMWV1VVJxVlJ0UTFELTN1S3NX``
```
Comment (ericenns) [inline docs-site/docs/extend/graphql.md:199]: **nitpick (blocking):** You have extra backticks here which is causing formatting errors when rendering the documentation.

**Review by ericenns — CHANGES_REQUESTED** [review]
This is starting to look good, but I have a few comments.

File: docs-site/docs/extend/graphql.md (hunk 2 above)
Comment (JeffreyThiessen) [reply docs-site/docs/extend/graphql.md]: Fixed.

File: docs-site/docs/intro.md
Comment (JeffreyThiessen) [reply docs-site/docs/intro.md:1]: Changed it back. I had changed it as the style guide in VSCode complained.

File: docs-site/docs/extend/graphql.md (hunk 1 above)
Comment (JeffreyThiessen) [reply docs-site/docs/extend/graphql.md:169]: Changed.

File: docs-site/docs/extend/access_tokens.png
Comment (JeffreyThiessen) [reply docs-site/docs/extend/access_tokens.png:1]: removed.

File: docs-site/docs/extend/edit_profile.png
Comment (JeffreyThiessen) [reply docs-site/docs/extend/edit_profile.png:1]: removed.

**Review by ericenns — APPROVED** [review]
Looks good to me!
